[Merged by Bors] - feat(geometry/euclidean/angle/oriented/right_angle): trigonometric lemmas

jsm28 opened the pull request:
Add lemmas about trigonometric functions in relation to right-angled triangles with oriented angles.  These are generally analogous to the unoriented case, but note that the property of an angle being a right angle is not symmetric with respect to reversing the order of the vectors / points in the oriented case (a positive right angle becomes a negative right angle).  Thus, one unoriented lemma tends to correspond to two lemmas in the oriented case, in order to state the trigonometric results for both angles other than the right angle, since those results are no longer so trivially equivalent.



---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

To indicate co-authors, include lines at the bottom of the commit message 
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]
-->

- [x] depends on: #17941

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

jsm28 committed:
feat(geometry/euclidean/angle/oriented/right_angle): trigonometric lemmas

Add lemmas about trigonometric functions in relation to right-angled
triangles with oriented angles.  These are generally analogous to the
unoriented case, but note that the property of an angle being a right
angle is not symmetric with respect to reversing the order of the
vectors / points in the oriented case (a positive right angle becomes
a negative right angle).  Thus, one unoriented lemma tends to
correspond to two lemmas in the oriented case, in order to state the
trigonometric results for both angles other than the right angle,
since those results are no longer so trivially equivalent.

eric-wieser commented on src/geometry/euclidean/angle/oriented/right_angle.lean:
```diff
@@ -0,0 +1,878 @@
+
+/-- The sine of an angle in a right-angled triangle multiplied by the hypotenuse equals the
+opposite side. -/
+lemma sin_oangle_left_mul_norm_of_oangle_eq_pi_div_two {p₁ p₂ p₃ : P} (h : ∡ p₁ p₂ p₃ = ↑(π / 2)) :
```
```suggestion
lemma sin_oangle_left_mul_dist_of_oangle_eq_pi_div_two {p₁ p₂ p₃ : P} (h : ∡ p₁ p₂ p₃ = ↑(π / 2)) :
```

jsm28 replied:
Filed #17941 with the corresponding fix for unoriented angle lemma naming.

jsm28 replied:
And merged in that branch and fixed the names here, thus marking this PR as depending on #17941.

jsm28 replied:
And merged in master now #17941 has been merged, so the naming fix is complete.

jsm28 committed:
fix(geometry/euclidean/angle/unoriented/right_angle): fix lemma naming

As pointed out by @eric-wieser in review of #17683 for corresponding
oriented angle lemmas, some lemma names refer to `norm` but should
refer to `dist`.

jsm28 committed:
Merge branch 'jsm28/right_angle_norm_dist_name_fix' into jsm28/oangle_right_angle

jsm28 committed:
Fix lemma naming

jsm28 committed:
Merge remote-tracking branch 'origin/master' into jsm28/oangle_right_angle

jcommelin reviewed (approved):
Thanks :tada:

bors merge
